Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -165,24 +179,24 @@ Writer Writer::Wrap(std::vector<uint8_t>* out_bytes) {
 // make it a bit easier to review. Refactor these to group the related methods
 // together (probably within their own file rather than here).
 
-void Writer::WriteTag(pb_wire_type_t wiretype, uint32_t field_number) {
-  bool status = pb_encode_tag(&stream_, wiretype, field_number);
-  if (!status) {
-    // TODO(rsgowman): figure out error handling
-    abort();
+Status Writer::WriteTag(pb_wire_type_t wiretype, uint32_t field_number) {
+  bool ok = pb_encode_tag(&stream_, wiretype, field_number);
+  if (!ok) {
+    return getInternalError(PB_GET_ERROR(&stream_));
   }
+  return Status::OK();
 }
 
-void Writer::WriteSize(size_t size) {
+Status Writer::WriteSize(size_t size) {
   return WriteVarint(size);
 }
 
-void Writer::WriteVarint(uint64_t value) {
-  bool status = pb_encode_varint(&stream_, value);
-  if (!status) {
-    // TODO(rsgowman): figure out error handling
-    abort();
+Status Writer::WriteVarint(uint64_t value) {
+  bool ok = pb_encode_varint(&stream_, value);
+  if (!ok) {
+    return getInternalError(PB_GET_ERROR(&stream_));
```

Comment:
I presume the memory pointed to by `PB_GET_ERROR` gets deallocated together with the stream?